src/classes/groundplaces.ts

```diff
@@ -1,66 +1,110 @@
+   * @param {CreateStopGroupProperties} stopGroupProperties - Properties that are needed to create a new StopGroup.
    * @param {StopClusterGpuid} stopClusterGpuid - Ground place unique identifier of the stopCluster parent.
-   * @param {StopGroup} stopGroupInfos - StopGroup informations.
-   * @returns {StopGroup}
+   * @returns {void}
    */
-  // @ts-ignore
-  public createStopGroup(stopClusterGpuid: StopClusterGpuid, stopGroupInfos: StopGroup): void {}
+  public createStopGroup(
+    segmentProviderStop: SegmentProviderStop,
+    stopGroupProperties: CreateStopGroupProperties,
+    stopClusterGpuid: StopClusterGpuid,
+  ): void {}
 
   /**
-   * @description Create the stopCluster with the values given.
-   * @param {StopCluster} stopClusterInfos - StopCluster informations.
-   * @returns {StopCluster}
+   * @description Create a new StopCluster from a StopGroup.
+   * @param {CreateStopClusterProperties} stopClusterProperties - Properties that are needed to create a new StopCluster.
+   * @returns {void}
    */
-  // @ts-ignore
-  public createStopCluster(stopClusterInfos: StopCluster): void {}
+  public createStopCluster(stopClusterProperties: CreateStopClusterProperties): void {}
 
   /**
    * @description Update the stopGroup with the new values given.
    * @param {StopGroupGpuid} stopGroupGpuid - Ground place unique identifier of a StopGroup.
-   * @param {StopGroupProperties} propertiesToUpdate - Properties that need to be update.
-   * @returns {StopGroup}
+   * @param {UpdateStopProperties} propertiesToUpdate - Properties that need to be update.
+   * @returns {void}
    */
-  // @ts-ignore
-  public updateStopGroup(stopGroupGpuid: StopGroupGpuid, propertiesToUpdate: StopGroupProperties): StopGroup {}
+  public updateStopGroup(stopGroupGpuid: StopGroupGpuid, propertiesToUpdate: UpdateStopProperties): void {
+    // Before make call to storageService
+    // Check if the rules are respected with the new properties to update
+    this.checkValidity();
+
+    // Then process to update
+    this.storageService.updatePlace(stopGroupGpuid, propertiesToUpdate);
+  }
 
   /**
    * @description Update the stopCluster with the new values given.
    * @param {StopClusterGpuid} stopClusterGpuid - Ground place unique identifier.
-   * @param {StopClusterProperties} propertiesToUpdate - Properties that need to be update.
-   * @returns {StopCluster}
+   * @param {UpdateStopProperties} propertiesToUpdate - Properties that need to be update.
+   * @returns {void}
    */
-  // @ts-ignore
-  public updateStopCluster(
-    stopClusterGpuid: StopClusterGpuid,
-    propertiesToUpdate: StopClusterProperties,
-  ): StopCluster {}
+  public updateStopCluster(stopClusterGpuid: StopClusterGpuid, propertiesToUpdate: UpdateStopProperties): void {
+    // Before make call to storageService
+    // Check if the rules are respected with the new properties to update
+    this.checkValidity();
```

Comment:
Check validity after the update :)

Reply:
We're having a discussion about this with Dimitri and Theo and we think that checking the rules should be done before any action. In this case it's not necessarily obvious but in another function that performs several actions (like createStop...), if we check that the rules are well respected at the end of the processing, what would happen if the first action performed works while the second one fails? We will find ourselves in a case where we have updated the groundplaces object but failed at the end. 

So we think that checkValidity should be perform before any actions performed on the storage

Do you confirm @rimiti @Theo-le-donne

Reply:
this is a non-sense. What is the purpose of checking validtiy of the file before an action ??
If the action has a bad side effect you will know the problem the next time you update somthing.

You have to : 
1. copy you storage 
2. execute the modifcation on the ground storage
3. check validity ON the global file. In case of validity error rollback thanks to the inital copy 

Otherwise you'll have supercomplicated functions to check all the rules according to the params & function called.

Reply:
A call has been made with Moussa to explain that. and go to the simplest solution.

Reply:
I've update the two function to have this new behavior, this should be better now